source/code/include/playfab/PlayFabCallRequestContainerBase.h

```diff
@@ -38,20 +39,23 @@ namespace PlayFab
         void SetRequestId(const std::string& newRequestId);
         std::string GetRequestBody() const;
         std::shared_ptr<PlayFabApiSettings> GetApiSettings() const;
+        std::shared_ptr<PlayFabAuthenticationContext> GetContext() const;
 
         /// <summary>
         /// This function is meant to handle logic of calling the error callback or success
         /// </summary>
         CallRequestContainerCallback GetCallback() const;
 
         void* GetCustomData() const;
+        virtual bool ValidateSettings() = 0;
```

Comment:
>virtual bool ValidateSettings() = 0; [](start = 8, length = 36)

Please do not add pure virtual methods to already released APIs. This is a bad practice in general. For an explanation - check out #20 here: [https://www.acodersjourney.com/top-25-cplusplus-api-design-mistakes-and-how-to-avoid-them/](https://www.acodersjourney.com/top-25-cplusplus-api-design-mistakes-and-how-to-avoid-them/)

Reply:
RIGHT! This was an added method!
Oh, yes, I see though... it makes this class abstract...
This one is hard. Let's discuss in person.

---
In reply to: [343286948](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r343286948) [](ancestors = 343286948)

Reply:
I'm going to investigate whether this method and all its requirements should just be in the base class.
That'll require moving some things from the child to the base here, if I decide it's relevant.
If I go that route, this change will basically still be added, but it won't be abstract.

---
In reply to: [343322808](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r343322808) [](ancestors = 343322808,343286948)

Reply:
Deb, you're right, ValidateSettings, and the settings, and the context should never have been in this file.
It's all removed.
Having PlayFab specifics of any kind in this file is a violation of the intent in the first place.
It's fully resolved now.

---
In reply to: [343371930](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r343371930) [](ancestors = 343371930,343322808,343286948)